internal/handlers/pg/pgdb/query_iterator.go

```diff
@@ -32,8 +32,8 @@ import (
 type queryIterator struct {
 	ctx         context.Context
 	rows        pgx.Rows
-	mxClosed    *sync.Mutex // protects closed
-	closed      bool        // indicates whether Close() was called
+	close       sync.Once   // to ensure that all Close() calls after the first will have no effect.
```

Comment:
```suggestion
	close       sync.Once   // to ensure that all Close() calls after the first will have no effect
```

We use dots only for complete and capitalized sentences.

no dots for sentence fragments

But the most idiomatic way would be to call that field `closeOnce` without any comment